### PR TITLE
fix serialize Struct, now Struct is not a collection

### PR DESCRIPTION
**Pull request by wildkain**
Correction of behavior during serialization Struct. Now Struct is perceived as a single object, not a collection.

### CODE REVIEW

**Comment by 0exp** [comment]
Kruto

**Comment by 0exp** [comment]
kogda?

File: spec/serializer_spec.rb [hunk 1]
```diff
@@ -210,6 +216,18 @@ class FooChildSerializer < FooSerializer
           it { is_expected.to eq expectation }
         end
       end
+
+      describe 'serialize Struct as single object' do
+        let(:person) { TestStruct.new('John', 'Dow') }
+        let(:expectation) { { name: 'John', other_param: 'Dow' } }
+        subject(:json) { TestStructSerializer.new(person).build_schema }
```
Comment (nesaulov) [inline spec/serializer_spec.rb]: it's a hash, not a json

File: spec/serializer_spec.rb [hunk 2]
```diff
@@ -210,6 +216,18 @@ class FooChildSerializer < FooSerializer
           it { is_expected.to eq expectation }
         end
       end
+
+      describe 'serialize Struct as single object' do
```
Comment (nesaulov) [inline spec/serializer_spec.rb]: 'serializing a single struct'

File: spec/serializer_spec.rb [hunk 3]
```diff
@@ -210,6 +216,18 @@ class FooChildSerializer < FooSerializer
...
+        let(:person) { TestStruct.new('John', 'Dow') }
+        let(:expectation) { { name: 'John', other_param: 'Dow' } }
+        subject(:json) { TestStructSerializer.new(person).build_schema }
+
+        it { is_expected.to eq expectation }
+
+        it 'raise surrealize instead surrealize_collection with struct' do
```
Comment (nesaulov) [inline spec/serializer_spec.rb]: let's move this example to a separate place, like `helper_spec.rb`

File: spec/serializer_spec.rb [hunk 4]
```diff
@@ -251,6 +269,23 @@ class FooChildSerializer < FooSerializer
         end
       end
     end
+
+    describe 'collection of Struct' do
```
Comment (nesaulov) [inline spec/serializer_spec.rb]: this example group should be inside `describe 'collection'` describe block (line 233)
and it should be named 'collection of structs'

File: spec/serializer_spec.rb [hunk 5]
```diff
@@ -251,6 +269,23 @@ class FooChildSerializer < FooSerializer
...
+      let(:person3) { TestStruct.new('Harley', 'Queen') }
+      let(:struct_collection) { [person1, person2, person3] }
+      let(:expectation) do
+        [
+          { name: 'John', other_param: 'Dow' },
+          { name: 'Mountain', other_param: 'Dew' },
+          { name: 'Harley', other_param: 'Queen' },
+        ]
+      end
+      subject(:json) { TestStructSerializer.new(struct_collection).build_schema }
```
Comment (nesaulov) [inline spec/serializer_spec.rb]: please add an empty line between lets and subject

File: spec/helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+TestStruct = Struct.new(:name, :other_param)
+
+RSpec.describe Surrealist::Helper do
+  describe 'serializing  a single struct' do
+    let(:person) { TestStruct.new('John', 'Dow') }
+
+    it 'raise surrealize instead surrealize_collection with struct' do
```
Comment (nesaulov) [inline spec/helper_spec.rb]: everything seems good, but the example message is quite misleading: nothing is raised, we are just asserting that `SurrealistHelper.collection?` returns false for a single struct
I suggest something like `specify 'a struct is not treated as a collection' do ...`

Comment (wildkain) [reply spec/helper_spec.rb]: fixed

**Pull request merged**